crypto/sparc_arch.inc

```diff
@@ -7,8 +7,8 @@
  * https://www.openssl.org/source/license.html
  */
 
-#ifndef OSSL_CRYPTO_SPARC_ARCH_H
-# define OSSL_CRYPTO_SPARC_ARCH_H
+#ifndef OSSL_CRYPTO_SPARC_ARCH_INC
+# define OSSL_CRYPTO_SPARC_ARCH_INC
```

Comment:
There is no reason for this rename, please rename it back

Reply:
I disagree.  These defines follow a _naming convention_.  Not renaming them break the convention.

Reply:
There is no reason to *rename the file*!

And now that I've see the rest of the changes, I think it should simply be moved to `include/crypto`, with the declaration of `OPENSSL_sparcv9cap_P` added to it.  Just like we have already done with `ppd_arch.h`

Reply:
+1 for doing that

Reply:
> And now that I've see the rest of the changes, I think it should simply be moved to `include/crypto`, with the declaration of `OPENSSL_sparcv9cap_P` added to it. Just like we have already done with `ppd_arch.h`

Unfortunately there is a reason. I've originally done exactly this and the build failed because the sparc_arch.h is included in assembler code which means the OPENSSL_sparcv9cap_P declaration cannot be added there.

If you have any idea how to solve that problem differently I am all for that.

Reply:
@levitte solved a little bit differently.